### PR TITLE
Fix #828: Path tracker crash handled when location services disabled

### LINKED ISSUE #828: App crashes in path tracker if location disabled
**Summary:** 

App crashes if path tracking is enabled while access to current location is disabled.

**Steps to reproduce:** 
Disable access to current location
Open app -> Nav Drawer -> Path tracker -> Click on the button in top right corner to start tracking

**Expected behavior:** 

User should be asked to enable location access.

**Screenshots:** 

path-tracker-crash

**Would you like to work on this?**
Yes

### PR DESCRIPTION
**Pull request by nb2998**
Fixes #828 
Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [X] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [X] If you have multiple commits please combine them into one commit by squashing them.

pathcrash

### CODE REVIEW

**Comment by nb2998** [comment]
@tarun0 @therajanmaurya Please review

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/pathtracking/PathTrackingActivity.java [hunk 1]
```diff
@@ -240,6 +249,50 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+        }
+        if (gpsEnabled && networkEnabled) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private void enableLocation() {
+        AlertDialog.Builder dialog = new AlertDialog.Builder(this,
```
Comment (tarun0) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/pathtracking/PathTrackingActivity.java]: It's recommended to use builder pattern for the sake of readability. i.e.

```
   new AlertDialog.Builder(...)
        .setTitle(...)
        .setMessage(...)
        ... ;
```

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/pathtracking/PathTrackingActivity.java [hunk 2]
```diff
@@ -240,6 +249,50 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+                        Intent myIntent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+                        startActivity(myIntent);
+                    }
+                });
+        dialog.setNegativeButton(getString(R.string.Cancel), new DialogInterface.OnClickListener() {
+
+            @Override
+            public void onClick(DialogInterface paramDialogInterface, int paramInt) {
+                paramDialogInterface.dismiss();
+
```
Comment (tarun0) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/pathtracking/PathTrackingActivity.java]: Unnecessary blank line.

File: mifosng-android/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -733,6 +733,12 @@
     <string name="incorrect_passcode">Incorrect Passcode</string>
     <string name="incorrect_passcode_more_than_three">You have entered incorrect Passcode more than 3 times. Please login again.</string>
     <string name="error_passcode">Passcode should be of 4 digit</string>
+    <string name="gps_network_not_enabled">GPS network is disabled</string>
+    <string name="open_location_settings">Open location settings</string>
+    <string name="Cancel">Cancel</string>
```
Comment (tarun0) [inline mifosng-android/src/main/res/values/strings.xml]: Are you sure it doesn't exist already?

Comment (tarun0) [inline mifosng-android/src/main/res/values/strings.xml]: Also, have a look at this interesting SO thread and the mentioned blog about the Location Manager vs Play Services.

Comment (tarun0) [reply mifosng-android/src/main/res/values/strings.xml]: Missed the link: https://stackoverflow.com/questions/33022662/android-locationmanager-vs-google-play-services

Comment (nb2998) [reply mifosng-android/src/main/res/values/strings.xml]: I googled it! I didn't know about this alternative method. Thanks, I am reading that!
Thanks for the review! I'll make the changes asap.

Comment (nb2998) [reply mifosng-android/src/main/res/values/strings.xml]: @tarun0 In that case, should [this](https://stackoverflow.com/questions/30754117/android-check-location-services-enabled-if-using-googleapiclient) be considered to check if location services are enabled? What do you suggest? Thanks!

**Comment by nb2998** [comment]
@tarun0 Made the changes.
Kindly let me know if I should follow [this](https://stackoverflow.com/questions/30754117/android-check-location-services-enabled-if-using-googleapiclient) instead of Location Manager to check locations services enabled or not.
Thanks!

**Comment by amogh-jrules** [comment]
Is this request going to be pulled in?
If not, I would be interested in working on a solution.

**Comment by amogh-jrules** [comment]
Is this request going to be pulled in?
If not, I would be interested in working on a solution.

**Comment by droidchef** [comment]
@nb2998 Please resolve the merge conflict and re-create a PR. 

Also please write a more descriptive commit message, describing your change in detail.

**Pull request closed without merging**